scripts/deploy-ceph-csi-operator.sh

```diff
@@ -0,0 +1,148 @@
+RBD_DRIVER_NAME="rbd.csi.ceph.com"
+CEPHFS_DRIVER_NAME="cephfs.csi.ceph.com"
+NFS_DRIVER_NAME="nfs.csi.ceph.com"
+
+# k8s csi sidecar image
+K8S_IMAGE_REPO=${K8S_IMAGE_REPO:-"registry.k8s.io/sig-storage"}
+
+TEMP_DIR="$(mktemp -d)"
+trap 'rm -rf "$TEMP_DIR"' EXIT
+
+function create_or_delete_imageset_configmap() {
+    local operation=$1
+    temp_file=$(mktemp "${TEMP_DIR}/imageset-configmap.XXXXXX.yaml")
+    cat <<EOF >"${temp_file}"
+apiVersion: v1
+kind: ConfigMap
+metadata:
+    name: ${IMAGESET_CONFIGMAP_NAME}
+    namespace: ${OPERATOR_NAMESPACE}
+data:
+    "plugin": "quay.io/cephcsi/cephcsi:canary"  # test image
+    "attacher": "${K8S_IMAGE_REPO}/csi-attacher:${CSI_ATTACHER_VERSION}"
+    "snapshotter": "${K8S_IMAGE_REPO}/csi-snapshotter:${CSI_SNAPSHOTTER_VERSION}"
+    "provisioner": "${K8S_IMAGE_REPO}/csi-provisioner:${CSI_PROVISIONER_VERSION}"
+    "registrar": "${K8S_IMAGE_REPO}/csi-node-driver-registrar:${CSI_NODE_DRIVER_REGISTRAR_VERSION}"
+    "resizer": "${K8S_IMAGE_REPO}/csi-resizer:${CSI_RESIZER_VERSION}"
+EOF
+    kubectl_retry "$operation" -f "${temp_file}"
+}
+
+function create_or_delete_encryption_configmap() {
+    local operation=$1
+    temp_file=$(mktemp "${TEMP_DIR}/encryption-configmap.XXXXXX.yaml")
+    cat <<EOF >"${temp_file}"
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: ${OPERATOR_NAMESPACE}
+  name: ${ENCRYPTION_CONFIGMAP_NAME}
+data:
+    config.json: ""
+EOF
+    kubectl_retry "$operation" -f "${temp_file}"
+}
+
+function create_or_delete_operator_config() {
+    # TODO: encryption config
+    local operation=$1
+    create_or_delete_imageset_configmap "$operation"
+    create_or_delete_encryption_configmap "$operation"
+
+    temp_file=$(mktemp "${TEMP_DIR}/operatorconfig.XXXXXX.yaml")
+    cat <<EOF >"${temp_file}"
+apiVersion: csi.ceph.io/v1alpha1
+kind: OperatorConfig
+metadata:
+    name: ceph-csi-operator-config
+    namespace: ${OPERATOR_NAMESPACE}
+spec:
+    driverSpecDefaults:
+        snapshotPolicy: volumeGroupSnapshot
+        controllerPlugin:
+            deploymentStrategy:
+                type: Recreate
+        generateOMapInfo: true
+        enableMetadata: true
+        log:
+            verbosity: 5 # csi pods log level
+        imageSet:
+            name: ${IMAGESET_CONFIGMAP_NAME}
+        encryption:
+            configMapName:
+                name: ${ENCRYPTION_CONFIGMAP_NAME}
+    log:
+        verbosity: 3 # operator log level
+EOF
+    kubectl_retry "$operation" -f "${temp_file}"
+}
+
+function deploy_or_delete_driver() {
+    local operator=$1
+    local driver_name=$2
+    temp_file=$(mktemp "${TEMP_DIR}/${driver_name}.XXXXXX.yaml")
+    cat <<EOF >"${temp_file}"
+apiVersion: csi.ceph.io/v1alpha1
+kind: Driver
+metadata:
+  name: ${driver_name}
+  namespace: ${OPERATOR_NAMESPACE}
+EOF
+    kubectl_retry "$operator" -f "${temp_file}"
+}
+
+function deploy_operator() {
+    kubectl_retry create -f "${OPERATOR_INSTALL}"
+    create_or_delete_operator_config "create"
```

Comment:
define a var for create and delete as its used in multiple places

Reply:
or use `create` and `delete` only once :wink:

```
    generate_operator_config
    generate_driver $RBD_DRIVER_NAME
    generate_driver $CEPHFS_DRIVER_NAME
    generate_driver $NFS_DRIVER_NAME
    kubectl_retry "create" -f "${TEMP_DIR}"
```

Please try to use the same coding standard everywhere, like `"${RBD_DRIVER_NAME}"` with the quotes and such.